realm/realm-library/src/objectServer/java/io/realm/internal/jni/OSJNIBsonProtocol.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.jni;
+
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+
+/**
+ * Protocol for passing {@link BsonValue}s to JNI.
+ *
+ * For now this just encapsulated the BSON value in a document with key {@value VALUE}. This
+ * overcomes the shortcoming of {@code org.bson.JsonWrite} not being able to serialize single values.
+ */
+public class OSJNIBsonProtocol {
```

Comment:
Normally we use the `Os` name to signal that the class is wrapping an existing C++ class, so using that name here is a bit confusing. I would probably just rename to `JniBsonProtocol`

Reply:
👍